core/shared/src/main/scala/monocle/function/Plated.scala

```diff
@@ -62,4 +65,19 @@ trait PlatedFunctions {
   def transformOf[A](l: Setter[A, A])(f: A => A)(a: A): A =
     l.modify(b => transformOf(l)(f)(f(b)))(a)
 
-}
+  /** transforming counting changes */
+  def transformC[A: Plated](pf: PartialFunction[A, A])(a: A): (Int, A) = {
```

Comment:
would you mind to replace `PartialFunction[A, A]` by `A => Option[A]` for consistency with `rewriteOf`